.github/workflows/ci.yml

```diff
@@ -40,3 +42,49 @@ jobs:
         run: coveralls --service=github
         env:
           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+  build:
+    name: Build
+    runs-on: ubuntu-latest
+    permissions:
+      contents: read
+    steps:
+      - uses: actions/checkout@v4
+      - name: Set up Python 3.10
+        uses: actions/setup-python@v4
+        with:
+          python-version: '3.10'
+          cache: 'pip'
+      -
+        name: Install build tools
+        run: |
+          pip3 --quiet install --upgrade build wheel
+      -
+        name: Build
+        run: |
+          python3 -m build .
+      -
+        uses: actions/upload-artifact@v3
+        with:
+            name: artifacts
+            path: dist/*
+            if-no-files-found: error
+            retention-days: 7
+
+  pypi-publish:
+    name: Publish
+    runs-on: ubuntu-latest
+    if: startsWith(github.ref, 'refs/tags/v')
```

Comment:
This job only runs if a new tag is pushed.  So that's only done after the manual intervention, changelog, and version file are updated.  PRs and general pushes to branches don't run this at all